src/main/java/io/stargate/sgv2/jsonapi/util/SortClauseUtil.java

```diff
@@ -8,10 +8,20 @@
 public class SortClauseUtil {
   public static List<FindOperation.OrderBy> resolveOrderBy(SortClause sortClause) {
     if (sortClause == null || sortClause.sortExpressions().isEmpty()) return null;
+    if (sortClause.sortExpressions().stream().findFirst().get().path().equals("$vector"))
+      return null;
     return sortClause.sortExpressions().stream()
         .map(
             sortExpression ->
                 new FindOperation.OrderBy(sortExpression.path(), sortExpression.ascending()))
         .collect(Collectors.toList());
   }
+
+  public static float[] resolveVsearch(SortClause sortClause) {
+    if (sortClause == null || sortClause.sortExpressions().isEmpty()) return null;
+    if (sortClause.sortExpressions().stream().findFirst().get().path().equals("$vector")) {
```

Comment:
Calculates `sortClause.sortExpressions().stream().findFirst().get()` twice, could assign result to a var, reuse.